acme/dns_challenge.go

```diff
@@ -108,13 +108,19 @@ func (s *dnsChallenge) Solve(chlng challenge, domain string) error {
 		timeout, interval = 60*time.Second, 2*time.Second
 	}
 
+	logf("[INFO][%s][timeout=%s][interval=%s] Checking DNS record propagation using %+v", domain, timeout, interval, RecursiveNameservers)
+
 	err = WaitFor(timeout, interval, func() (bool, error) {
 		return PreCheckDNS(fqdn, value)
 	})
 	if err != nil {
 		return err
 	}
 
+	extraWaitTime := 20 * time.Second
+	logf("[INFO][%s] DNS record is live, inserting %s of wait time to reduce the probability of a Let's Encrypt 400 error.", domain, extraWaitTime)
+	time.Sleep(extraWaitTime)
+
 	return s.validate(s.jws, domain, chlng.URI, challenge{Resource: "challenge", Type: chlng.Type, Token: chlng.Token, KeyAuthorization: keyAuth})
```

Comment:
If might make more sense to put a for loop around this and __then__ return so we can catch the crappy 400 errors from Let's Encrypt.